atom/browser/api/atom_api_web_contents.cc

```diff
@@ -451,10 +460,20 @@ void WebContents::DidGetResourceResponseStart(
 void WebContents::DidGetRedirectForResourceRequest(
     content::RenderFrameHost* render_frame_host,
     const content::ResourceRedirectDetails& details) {
+  v8::Locker locker(isolate());
+  v8::HandleScope handle_scope(isolate());
+
+  v8::Local<v8::Value> response_headers = mate::ConvertToV8(isolate(),
```

Comment:
You don't have to do the conversion manually, you can just call `Emit(..., details.headers.get())` for automatic conversion, and the `Locker` and `HandleScope` calls can also be removed since `Emit` has taken care of it.

The `DidGetResourceResponseStart` needs to do the same change too.